packages/hydrogen/src/pagination/Pagination.ts

```diff
@@ -18,41 +18,42 @@ type PaginationState = {
   pageInfo: PageInfo | null;
 };
 
-type Props<Resource extends Connection> = {
-  connection: Resource;
-  autoLoadOnScroll?: boolean | IntersectionOptions;
-};
-
 interface PaginationInfo {
-  endCursor: Maybe<string> | undefined;
+  state: {
+    nodes: ProductConnection['nodes'] | any[];
```

Comment:
Is it possible to remove the `any`?

Reply:
I'm not sure why it is there in the first place. I'll give it a shot.

Reply:
Resolved, removed all references to ProductConnection, and instead infer the nodes type based on the connection prop passed to `<Pagination>`